src/runtime_src/core/common/device_core.h

```diff
@@ -200,6 +203,10 @@ class device_core {
     } QueryRequest;
 
     virtual void query_device(uint64_t _deviceID, QueryRequest _eQueryRequest, const std::type_info & _typeInfo, boost::any &_returnValue) const = 0;
+    virtual void get_IpLayout(uint64_t _deviceID, XU_IP_LAYOUT **ipLayout, DWORD size) const = 0;
```

Comment:
XU_... are windows driver specific.  Replace with structures from xclbin.h
Please don't use camel case for function names, stay consistent, how about get_ip_layout.